docs/cql/cql-extensions.md

```diff
@@ -232,6 +232,25 @@ to explicitly mark them as such.
 
 ## Expressions
 
+### NULL
+
+Scylla aims for a uniform handling of NULL values in expressions, inspired
+by SQL: The overarching principle is that a NULL signifies an absense of
+value, so any expression calculated based on a NULL also results in a NULL.
+For example, the result of `x + NULL`, `x = NULL`, `x < NULL` are all NULL,
+no matter what `x` is. Even the expression `NULL = NULL` returns NULL, not
+TRUE.
+
+From the last example we can see that the `SELECT` filter `WHERE x = NULL`
+matches (returns `TRUE`) no row. It doesn't even match rows where x is
+missing. To match rows with missing x, SQL offers the syntax `x IS NULL`
+(and similarly, also `x IS NOT NULL`). Scylla does not yet implement this
+syntax.
+
+In contrast, Cassandra is less consistent in its handling of nulls.
+The example `x = NULL` is considered an error, not a valid expression
+whose result is NULL.
+
```

Comment:
There could also be a note about `AND`. An example like `TRUE AND NULL` is `NULL`, but `TRUE AND FALSE AND NULL` is `FALSE`.

Reply:
`TRUE AND FALSE AND NULL` is `FALSE` I guess because of short-circuiting? Sort of like in C, `0 && abort()` is false, not aborting? Weird, but understandable I guess.

Reply:
It's consistent with the notion that NULL means an absence of a value, which can be interpreted as an unknown value.
`FALSE AND NULL` will return `FALSE` because no matter what value the `NULL` might be, the result will still be `FALSE`.
The behavior and its justification was described in #12300.

Reply:
I also confirmed that PostgresQL uses the same rules.